### ISSUE TITLE
unable to login - Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by thezoggy** [Sonarr/Sonarr#1934]
friends sonarr instance got hacked/someone put a pw on it..  went to recover it by following ( https://forums.sonarr.tv/t/http-password-reset/4498 ) to change the login auth to None. then got back in and set the pw he wanted... when I try logging into his instance I see:
```
{
  "message": "Object reference not set to an instance of an object",
  "description": "System.NullReferenceException: Object reference not set to an instance of an object\n  at Nancy.Authentication.Forms.FormsAuthentication.EncryptAndSignCookie (System.String cookieValue, Nancy.Authentication.Forms.FormsAuthenticationConfiguration configuration) [0x00000] in <91c66dca293e4531909364de82529d4c>:0 \n  at Nancy.Authentication.Forms.FormsAuthentication.BuildCookie (System.Guid userIdentifier, System.Nullable`1[T] cookieExpiry, Nancy.Authentication.Forms.FormsAuthenticationConfiguration configuration) [0x0000d] in <91c66dca293e4531909364de82529d4c>:0 \n  at Nancy.Authentication.Forms.FormsAuthentication.UserLoggedInRedirectResponse (Nancy.NancyContext context, System.Guid userIdentifier, System.Nullable`1[T] cookieExpiry, System.String fallbackRedirectUrl) [0x001ea] in <91c66dca293e4531909364de82529d4c>:0 \n  at Nancy.Authentication.Forms.ModuleExtensions.LoginAndRedirect (Nancy.INancyModule module, System.Guid userIdentifier, System.Nullable`1[T] cookieExpiry, System.String fallbackRedirectUrl) [0x00006] in <91c66dca293e4531909364de82529d4c>:0 \n  at NzbDrone.Api.Authentication.AuthenticationModule.Login (NzbDrone.Api.Authentication.LoginResource resource) [0x00166] in M:\\BuildAgent\\work\\b69c1fe19bfc2c38\\src\\NzbDrone.Api\\Authentication\\AuthenticationModule.cs:46 \n  at NzbDrone.Api.Authentication.AuthenticationModule.<.ctor>b__2_0 (System.Object x) [0x00000] in M:\\BuildAgent\\work\\b69c1fe19bfc2c38\\src\\NzbDrone.Api\\Authentication\\AuthenticationModule.cs:21 \n  at (wrapper dynamic-method) System.Object:CallSite.Target (System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)\n  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 "
}
```

had to nuke all cookies to get it working again

**Comment by githubsean** [comment]
I found that the Forms based login is not working - I changed it to Basic, and it seems to be working.
I think this broke either after a recent Sonarr update, or after the big Mono update the other day.

**Comment by markus101** [comment]
Which version of mono are/did you see this with?

This is failing within a library that hasn't been upgraded in some time, unless it's related to the cookie path change (though we didn't see this same issue in develop), so perhaps an issue since mono 5.0 was released, that goes away once the cookies are cleared?

**Comment by thezoggy** [comment]
this was on keith's box..
```
$ mono -V
Mono JIT compiler version 4.6.2 (Stable 4.6.2.16/ac9e222 Tue Jan  3 11:48:26 UTC 2017)
```

**Comment by githubsean** [comment]
This is me:
```
sean@htpc:~$ mono -V
Mono JIT compiler version 4.8.1 (Stable 4.8.1.0/22a39d7 Tue May  2 22:33:18 UTC 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  x86
        Disabled:      none
        Misc:          softdebug
        LLVM:          supported, not enabled.
        GC:            sgen
```
Ubuntu 16.04

**Comment by githubsean** [comment]
I think I have worked out what was happening. I use nginx providing TLS/SSL with sonarr running behind it.
I have the URL base set to /tv. The problem was with the nginx config which was working before a recent change.
Before, the following worked - note "sonarr" is defined by an earlier "upstream" directive.
```
  location /tv {
    proxy_pass              http://sonarr/tv;
    proxy_cookie_path / /tv/ ;
    client_max_body_size 10m;
  }
```
When I traced what was happening using Chrome's debug, I could see the problem was with the path used by the cookie that is set when the login button is pressed. I removed the "proxy_cookie_path" directive, and things started to work.
The original intention of the proxy_cookie_path was to ensure that if Sonarr tried to set a cookie with a path of "/" it would get changed to "/tv/". I have several servers on different end-points, and I was getting collissions in setting cookies by different apps.
It now seems that Sonarr has been changed such that this is no longer necessary - or at least, removing it, keeping it simple, fixes things.
```
  location /tv {
    proxy_pass              http://sonarr/tv;
    client_max_body_size 10m;
  }
```
